src/actions/wapeSale/getSaleStatus.ts

```diff
@@ -0,0 +1,29 @@
+import * as ethers from "ethers";
+import { WapeSale } from "../../contracts/types";
+import { SaleStatus } from "../../types";
+
+export const getSaleStatus = async (contract: WapeSale) => {
+  const saleStarted = await contract.saleStarted();
+
+  if (!saleStarted) {
+    return SaleStatus.NotStarted;
+  }
+
+  const currentBlock = await contract.provider.getBlockNumber();
+  const startBlock = await contract.saleStartBlock();
```

Comment:
I am guessing in this function calls 4 times of view functions, and it will take much time. 
What about using `multicall` or `Promise.all` of 4 different calling?